### PR TITLE
Lesson 2_Submit Assignments

### PR DESCRIPTION
**Pull request by codeeMadness**
Hi thay Quang,

There are still one quiz left (quiz 6 - quite tough!), I'm still working on it.
At quiz 2, I do not feel satisfied for my solution, so I will look for more better one.
In the meanwhile, please kindly help to review some other assignments I have resolved.

Regards,
Hang Nguyen

### CODE REVIEW

File: Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    def distance(self, x, y):
+        return x ** 2 + y ** 2
+
+    def kClosest(self, points, k):
+        dict = {}
+        for i in range(len(points)):
+            dict[i] = self.distance(points[i][0], points[i][1])
+        
+        heap = list(dict.values())
+        heapq.heapify(heap)
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py]: O(n)

File: Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        closest = []
+        while k > 0:
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py]: O(klogn)

File: Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            closestDistance = heapq.heappop(heap)
+            closest.append(closestDistance)
+            k -= 1
+        
+        output = []
+        for index, distance in dict.items():
```
Comment (quanghoang33) [inline Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py]: O(nxk)

Comment (quanghoang33) [reply Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py]: QSelect: O(n) average

File: Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py (hunk 2 above)
Comment (quanghoang33) [reply Homework/Hang_Nguyen/lesson_2/2.k-closest-points-to-origin.py]: {dist, index}

**Review by quanghoang33 — CHANGES_REQUESTED** [review]

**Comment by codeeMadness** [comment]
Hi thay @quanghoang33 
Please kindly help to review my assignment again.
Thanks,
Hang Nguyen

**Comment by quanghoang33** [comment]
lgtm

**Pull request merged**